parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -29,27 +30,21 @@ pub use weights::WeightInfo;
 	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
 )]
 #[scale_info(skip_type_params(M))]
-#[codec(mel_bound(SourceId: MaxEncodedLen))]
-pub struct Message<SourceId, M: Get<u32>>
-where
-	SourceId: Parameter + Member + MaxEncodedLen,
-{
+pub struct Message<M: Get<u32>> {
 	/// ID of source parachain
-	source_id: SourceId,
+	parachain_id: ParaId,
 	/// Unique nonce to prevent replaying messages
 	#[codec(compact)]
 	nonce: u64,
+	// TODO: add handler: u16?
 	/// Payload for target application.
 	payload: BoundedVec<u8, M>,
 }
 
-impl<SourceId, M: Get<u32>> Into<Token> for Message<SourceId, M>
-where
-	SourceId: Decode + Parameter + Member + MaxEncodedLen, //+ TypeInfo,
-{
+impl<M: Get<u32>> Into<Token> for Message<M> {
 	fn into(self) -> Token {
 		Token::Tuple(vec![
-			Token::Bytes(self.source_id.encode()),
+			Token::Bytes(self.parachain_id.encode()),
```

Comment:
Parachain ID is actually an ABI-encoded unsigned u32. That's what the ethereum side is expecting:
 1. https://github.com/Snowfork/snowbridge/blob/0acc8f2dc7d36b9ac008d3db84856d91a4198a82/core/packages/contracts/src/Types.sol#L4
2. https://github.com/Snowfork/snowbridge/blob/0acc8f2dc7d36b9ac008d3db84856d91a4198a82/core/packages/contracts/src/InboundQueue.sol#L36
```suggestion
			Token::Uint(self.origin.into()),
```